src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py

```diff
@@ -196,39 +209,63 @@ def update_progress(cur_loss, iteration):
+            train_batch.data[0], ctx_list=ctx, batch_axis=0)
+        train_batch_label = _mx.gluon.utils.split_and_load(
+                train_batch.label[0], ctx_list=ctx, batch_axis=0)
+        train_outputs = []
 
+        # Inside training scope
         with _autograd.record():
-            output = model(data)
-            loss = softmax_cross_entropy(output, label)
-        loss.backward()
-        # update parameters
-        trainer.step(1)
+            for x, y in zip(train_batch_data, train_batch_label):
+                z = model(x)
+                # Computes softmax cross entropy loss.
+                loss = softmax_cross_entropy(z, y)
+                # Backpropagate the error for one iteration.
+                loss.backward()
+                train_outputs.append(z)
+            # Compute training accuracy
+            train_accuracy.update(train_batch_label, train_outputs)
+
+        # Compute validation accuracy
+        for validation_batch in validation_loader:
+            validation_batch_data = _mx.gluon.utils.split_and_load(
+                validation_batch.data[0], ctx_list=ctx, batch_axis=0)
+            validation_batch_label = _mx.gluon.utils.split_and_load(
+                validation_batch.label[0], ctx_list=ctx, batch_axis=0)
+            validation_outputs = []
+            for x, y in zip(validation_batch_data, validation_batch_label):
+                z = model(x)
+                validation_outputs.append(z)
+            validation_accuracy.update(validation_batch_label, validation_outputs)
+
+        # Make one step of parameter update. Trainer needs to know the
+        # batch size of data to normalize the gradient by 1/batch_size.
+        trainer.step(train_batch.data[0].shape[0])
         # calculate training metrics
-        cur_loss = loss.mean().asscalar()
+        train_loss = loss.mean().asscalar()
+        train_time = _time.time() - start_time
         
-        update_progress(cur_loss, batch.iteration)
-        iteration = batch.iteration
+        if verbose and train_batch.iteration > iteration:
+            iteration = train_batch.iteration
+            table_printer.print_row(iteration=iteration,
+                                    train_loss=float(train_loss),
+                                    train_accuracy=train_accuracy.get()[1],
+                                    validation_accuracy=validation_accuracy.get()[1],
+                                    time=train_time)
 
-    training_time = _time.time() - start_time
-    if verbose:
-        print(hr)   # progress table footer
     state = {
         '_model': model,
         '_class_to_index': class_to_index,
         'num_classes': len(classes),
         'classes': classes,
         'input_image_shape': (1, BITMAP_WIDTH, BITMAP_HEIGHT),
         'batch_size': batch_size,
-        'training_loss': cur_loss,
-        'training_time': training_time,
+        'training_loss': train_loss,
```

Comment:
Shouldn't we also include the training accuracy, and validation accuracy if available?

Reply:
Fixed.